cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -254,25 +269,63 @@ template <typename T = uint8_t>
       next_chunk++;
     }
     if (io_size != 0) {
-      auto& source = sources[chunk_source_map[chunk]];
-      if (source->is_device_read_preferred(io_size)) {
-        auto buffer        = rmm::device_buffer(io_size, stream);
-        auto fut_read_size = source->device_read_async(
-          io_offset, io_size, static_cast<uint8_t*>(buffer.data()), stream);
+      auto& source              = sources[chunk_source_map[chunk]];
+      auto const is_device_read = source->is_device_read_preferred(io_size);
+      reads.push_back({io_size, io_offset, is_compressed, is_device_read, chunk, next_chunk});
+      if (is_device_read) {
+        if (is_compressed) {
+          compressed_read_size += io_size;
+        } else {
+          uncompressed_read_size += io_size;
+        }
+      }
+    }
+    chunk = next_chunk;
+  }
+
+  auto uncompressed_buffer = [&]() {
+    return uncompressed_read_size > 0 ? rmm::device_buffer(uncompressed_read_size, stream)
+                                      : rmm::device_buffer(0, stream);
+  }();
+
+  auto compressed_buffer = [&]() {
+    return compressed_read_size > 0 ? rmm::device_buffer(compressed_read_size, stream)
+                                    : rmm::device_buffer(0, stream);
+  }();
+
+  auto compressed_buffer_data   = static_cast<uint8_t*>(compressed_buffer.data());
+  auto uncompressed_buffer_data = static_cast<uint8_t*>(uncompressed_buffer.data());
```

Comment:
I think you're getting a `void *` back from the `.data()` call so you could probably use `std::byte *` here? Trying to push for `std::byte` adoption where possible.

Reply:
Also applies to a few places below.

Reply:
Recommend to use `reinterpret_cast`.